prototype/frameworks/llamastack/kubernetes/README.md

```diff
@@ -0,0 +1,80 @@
+# VLLM serve
+This serves as a starting point to deploy `meta-llama/Llama-3.1-8B-Instruct` long term this should be handled by RHOAI.
+
+## Requirements
+A secret must be created to allow for the vLLM server to pull in the model at startup time.
+
+```
+oc create secret generic huggingface-secret --from-literal=HF_TOKEN=hf_values
+```
+
+## Deploy
+The vLLM deployment will take the secret from above and use it to pull in the model. To launch the vLLM deployment run the following.
+
+```
+oc create -f llama-serve/vllm.yaml
+oc expose deployment/vllm
+```
+
+## Using the vLLM server
+The vLLM instance is running with a service this service can be accessed by other namespaces in the cluster but also by using port-forward.
+
+To use port-forward run the following.
+
+```
+kubectl port-forward service/vllm 8000:8000
+```
+
+## Testing
+To test the model is operating correctly.
+
+```
+ curl http://localhost:8000/v1/completions \
+    -H "Content-Type: application/json" \
+    -d '{
+        "model": "meta-llama/Llama-3.1-8B-Instruct",
+        "prompt": "San Francisco is a",
+        "max_tokens": 200,
+        "temperature": 0
+    }'
+```
+
+
+# Llama stack
+To run llama stack we can utlize the kubernetes svc address along with providing a configmap for the unique values we want to test.
+
+## Configmap
+A predefined configmap exists with entirely too many options but is a placeholder for what can be added. Create the configmap defining your unique settings. Then create the CM.
+
+```
+oc create -f llama-stack/configmap.yaml
+```
+
+
+## Deployment
+The deployment doesn't really have any unique values at this time. We will most likely learn more as we progress
+
+```
+oc create -f llama-stack/deployment.yaml
+```
+
+## Access
+This deployment isn't exposed so depending on where you need to access it from expose it accordingly.
+
+If you want to deploy a chat server and point to the llama-stack then run the following.
+
+```
+oc expose deployment/llamastack-deployment
+```
+
+If you want to access the llama-stack externally from the cluster then you must expose the svc which will create a route.
+
+```
+oc expose svc/llamastack-deployment
+```
+
+This will generate a route similar to this.
+
+```
+llamastack-deployment-llama-serve.apps.ocp-beta-test.nerc.mghpcc.org
+```
```

Comment:
Could you also mention the port to be used? And can we also add a section at the end like so:
 ### Testing the Llamastack Server
In order to test the Llamastack server, you can try some of the examples mentioned [here](https://github.com/redhat-et/agent-frameworks/tree/main/prototype/frameworks/llamastack/scripts) by setting the following env vars:
```
INFERENCE_MODEL="meta-llama/Llama-3.1-8B-Instruct"
LLAMA_STACK_PORT= <mention the port number>
```

When connecting to the server using `LlamaStackClient` make sure to update the `base_url` with the URL of the vLLM server you deployed.